### PR TITLE
Use k8s-test-infra devel image

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -150,7 +151,7 @@ build-image: $(DOCKERFILE_DEVEL)
 		--build-arg MOQ_VERSION="$(MOQ_VERSION)" \
 		--tag $(BUILDIMAGE) \
 		-f $(DOCKERFILE_DEVEL) \
-		.
+		$(K8S_TEST_INFRA)
```
Comment (elezar) [inline Makefile:154]: This still doesn't allow us to customize the docker image with other dependencies, correct?

Comment (ArangoGutierrez) [reply Makefile:154]: Correct. The whole purpose of https://github.com/NVIDIA/k8s-dra-driver/pull/83 was to enable local builds with specific versions, but no extra dependencies. 
This PR simply makes sure we continue to hold a central Dockerfile for Build environments, but allows to perform local builds with _custom_ dep versions

File: Makefile [hunk 2]
```diff
@@ -36,7 +36,8 @@ TARGETS := $(MAKE_TARGETS) $(CMD_TARGETS)
 
 DOCKER_TARGETS := $(patsubst %,docker-%, $(TARGETS))
 .PHONY: $(TARGETS) $(DOCKER_TARGETS)
-DOCKERFILE_DEVEL := $(CURDIR)/deployments/container/Dockerfile.devel
+DOCKERFILE_DEVEL := "images/devel/Dockerfile"
```
Comment (klueska) [inline Makefile:39]: Is there supposed to be a file added for this?:
```
+ DOCKERFILE_DEVEL := "images/devel/Dockerfile"
```

File: Makefile (hunk 1 above)
Comment (klueska) [inline Makefile:154]: This variable will be replaced as `"https://github.com/NVIDIA/k8s-test-infra.git"`.
Is that a valid context for docker?

File: Makefile (hunk 2 above)
Comment (ArangoGutierrez) [reply Makefile:39]: no, this is the location of the file at the `K8S_TEST_INFRA` location

File: Makefile (hunk 1 above)
Comment (ArangoGutierrez) [reply Makefile:154]: yes, Docker supports remote Dockerfiles in Git repos

**Comment by ArangoGutierrez** [comment]
last week @elezar and I had a quick ad-hock conversation on whether it makes sense to continue to push for a centralized image (the k8s-test-infra) image, or we should roll back the per-repo devel image, just with some rules so it looks similar across all our repos. So far the centralized image has added no value, and perhaps has added complications like the one leading to this PR.

If @elezar / @klueska think we should roll back, I'll have it as a TODO and prepare a per-repo solution.

File: Makefile (hunk 1 above)
Comment (klueska) [reply Makefile:154]: Ack

File: Makefile [hunk 3]
```diff
@@ -36,7 +36,8 @@ TARGETS := $(MAKE_TARGETS) $(CMD_TARGETS)
...
+K8S_TEST_INFRA := "https://github.com/NVIDIA/k8s-test-infra.git"
```
Comment (klueska) [inline Makefile:40]: Should these be `?=` so we can overwrite them when invoking `make` (if desired)?

Comment (klueska) [inline Makefile:40]: Should this variable be called `DOCKERFILE_CONTEXT"

**Comment by klueska** [comment]
> last week @elezar and I had a quick ad-hock conversation on whether it makes sense to continue to push for a centralized image (the k8s-test-infra) image, or we should roll back the per-repo devel image, just with some rules so it looks similar across all our repos. So far the centralized image has added no value, and perhaps has added complications like the one leading to this PR.
> 
> If @elezar / @klueska think we should roll back, I'll have it as a TODO and prepare a per-repo solution.

I think a centralized one is fine as long as it's configurable for each repo.

**Comment by elezar** [comment]
> I think a centralized one is fine as long as it's configurable for each repo.

My point was that our repos represent a disjoint set of components with different requirements and lifecycles. Trying to consolidate tooling in a central location may be effort that doesn't serve a direct purpose at present. Adding per-repo customization removes the primary benefit of this which is image reuse since a local image needs to be built for non-standard requirements. Furthermore, a user may not be aware of the fact that they NEED to build a custom image for local development since this is not explicitly encoded in the makefile.

Looking at the current devel image's Dockerfile the use of `moq`, `golangci-lint`, and setting the git config options on the `/work` directory are the only three things that are (somewhat) universally applicable. The other tooling was added for projects such as the `k8s-dra-driver` (although it may be used in some other projects). When considering codebases such as `go-nvml` the tooling becomes more specific and it may not make sense to centrally manage these.

**Comment by klueska** [comment]
I don't feel strongly either way.

On one hand, I like the per-repo images that we had originally because I could just update them as necessary for the particular project at hand, without worrying that I might "break" some other project (or become out of sync with some centralized image). This image is only meant to be used locally for build / devel purposes, so it makes sense to also define / build it locally instead of pulling it from some centralized location.

On the other hand, I can see how it can quickly become hard to keep the set of local Dockerfiles we have consistent across repos (for the parts of them that can/should be kept consistent).

Does it then make sense to have a "base" image hosted in k8s-test-infra that only contains the base set of tooling we know all devel images should have and then have each repo define a local Dockerfile that uses this as a base?

**Comment by ArangoGutierrez** [comment]
Rebased and addressed comments

**Review by klueska — APPROVED** [review]
Let's merge this for now. We should continue to discuss whether we want a local vs. remote image, but the default build is broken at the moment and we need something that will fix it.

**Pull request merged**